### ISSUE TITLE
vsphere: Multiple cores per vcpu

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-vsphere#42]
_This issue was originally opened by @datroup as hashicorp/terraform#8446. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi,

I cannot find this in any of the documentation, but when creating vm's in vsphere with terraform we're unable to set the number of cores per vcpu. Is this possible, or can this be a feature request? I know from a technical point of view it doesn't matter but we have a requirement for this to be set.

D

**Comment by jason-azze** [comment]
I would like to give a hearty second to this enhancement request, and I can cite a strong technical motivation.

If I use TF to provision a Windows 7 machine with, let's say, 16 vCPUs -- I'll end up with 16 sockets with 1 core each. The problem is, the Windows 7 OS can only see 2 sockets and therefore only 2 of those cores. (I just discovered this the hard way when a Windows compile project took 14 hours to complete instead of 2). 
![one-core-sixteen-sockets](https://user-images.githubusercontent.com/1788669/32954415-0394d34a-cb81-11e7-9998-7a6db2e35f6b.png)

It looks like the default in resource_vsphere_virtual_machine.go is 1. It would be wonderful to expose NumCoresPerSocket as a configurable parameter.
```golang
  // make config spec
  configSpec := types.VirtualMachineConfigSpec{
    Name:              vm.name,
    NumCPUs:           vm.vcpu,
    NumCoresPerSocket: 1,
    MemoryMB:          vm.memoryMb,
    MemoryAllocation: &types.ResourceAllocationInfo{
      Reservation: &vm.memoryAllocation.reservation,
    },
    Flags: &types.VirtualMachineFlagInfo{
      DiskUuidEnabled: &vm.enableDiskUUID,
    },
    Annotation: vm.annotation,
  }
```
I just took a peek at the branch for the big refactor release (circa November 2017), and it looks like 1 is still hard-coded.

I need to learn golang so I can start helping more directly.
